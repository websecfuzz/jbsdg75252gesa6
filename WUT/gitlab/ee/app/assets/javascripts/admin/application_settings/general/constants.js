import { s__ } from '~/locale';

export const DROPZONE_DESCRIPTION_TEXT = s__(
  'Licenses|Drag your license file here or %{linkStart}select a file to upload%{linkEnd}.',
);
export const FILE_UPLOAD_ERROR_MESSAGE = s__('Licenses|The file could not be uploaded.');
export const FILE_DROP_ERROR_MESSAGE = s__(
  'Licenses|Error: You are trying to upload something other than a file',
);
export const DROP_TO_START_MESSAGE = s__('Licenses|Drop your license file to start the upload.');
