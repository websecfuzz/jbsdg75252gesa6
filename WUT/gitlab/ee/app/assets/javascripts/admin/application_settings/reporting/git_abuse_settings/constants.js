import { s__, __, sprintf } from '~/locale';

export const MIN_NUM_REPOS = 0;
export const MAX_NUM_REPOS = 10000;

export const MIN_TIME_PERIOD = 0;
export const MAX_TIME_PERIOD = 10 * 24 * 60 * 60; // 10 days in seconds;

export const MAX_ALLOWED_USERS = 100;

export const MIN_ALERTED_USERS = 1;
export const MAX_ALERTED_USERS = 100;

export const NUM_REPOS_BLANK_ERROR = s__(
  "GitAbuse|Number of repositories can't be blank. Set to 0 for no limit.",
);
export const NUM_REPOS_NAN_ERROR = s__('GitAbuse|Number of repositories must be a number.');
export const NUM_REPOS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Number of repositories should be between %{minNumRepos}-%{maxNumRepos}.`),
  {
    minNumRepos: MIN_NUM_REPOS,
    maxNumRepos: MAX_NUM_REPOS,
  },
);

export const TIME_PERIOD_BLANK_ERROR = s__(
  "GitAbuse|Reporting time period can't be blank. Set to 0 for no limit.",
);
export const TIME_PERIOD_NAN_ERROR = s__('GitAbuse|Reporting time period must be a number.');
export const TIME_PERIOD_LIMIT_ERROR = sprintf(
  s__(
    `GitAbuse|Reporting time period should be between %{minTimePeriod}-%{maxTimePeriod} seconds.`,
  ),
  {
    minTimePeriod: MIN_TIME_PERIOD,
    maxTimePeriod: MAX_TIME_PERIOD,
  },
);

export const ALLOWED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|You cannot specify more than %{maxAllowedUsers} excluded users.`),
  {
    maxAllowedUsers: MAX_ALLOWED_USERS,
  },
);

export const NUM_REPO_LABEL = s__('GitAbuse|Number of repositories');
export const NUM_REPO_DESCRIPTION = s__(
  "GitAbuse|The maximum number of unique repositories a user can download in the specified time period before they're banned.",
);
export const REPORTING_TIME_PERIOD_LABEL = s__('GitAbuse|Reporting time period (seconds)');

export const ALLOWED_USERS_LABEL = s__('GitAbuse|Excluded users');
export const ALLOWED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are excluded from the Git abuse rate limit.',
);

export const ALERTED_USERS_LABEL = s__('GitAbuse|Send notifications to');
export const ALERTED_USERS_DESCRIPTION = s__(
  'GitAbuse|Users who are emailed when Git abuse rate limit is exceeded.',
);
export const ALERTED_USERS_LIMIT_ERROR = sprintf(
  s__(`GitAbuse|Select between %{minAlertedUsers} and %{maxAlertedUsers} users to notify.`),
  {
    minAlertedUsers: MIN_ALERTED_USERS,
    maxAlertedUsers: MAX_ALERTED_USERS,
  },
);

export const AUTO_BAN_TOGGLE_LABEL = s__(
  'GitAbuse|Automatically ban users from this %{scope} when they exceed the specified limits',
);

export const SEARCH_USERS = __('Search users');
export const SEARCH_TERM_TOO_SHORT = __('Enter at least three characters to search');
export const NO_RESULTS = __('No results');

export const SAVE_CHANGES = __('Save changes');
export const SUCCESS_MESSAGE = __('Application settings saved successfully.');

export const LOAD_ERROR_MESSAGE = __(
  'An error occurred while saving your settings. Try saving them again.',
);
export const SAVE_ERROR_MESSAGE = __(
  'An error occurred while retrieving your settings. Reload the page to try again.',
);
