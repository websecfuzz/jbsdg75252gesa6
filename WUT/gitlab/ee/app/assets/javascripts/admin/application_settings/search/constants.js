import { __, s__ } from '~/locale';

export const MINIMUM_QUERY_LENGTH = 3;
export const NO_RESULTS_TEXT = __('No results found.');
export const SEARCH_QUERY_TOO_SHORT = __('Enter at least three characters to search.');
export const ENTITIES_FETCH_ERROR = s__(
  'AdvancedSearch|Could not fetch index entities. Please try again later.',
);
