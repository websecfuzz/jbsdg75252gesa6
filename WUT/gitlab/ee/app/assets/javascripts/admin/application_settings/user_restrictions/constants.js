import { helpPagePath } from '~/helpers/help_page_helper';

export const PRIVATE_PROFILES_DISABLED_ICON = 'private-profiles-disabled-icon';
export const PRIVATE_PROFILES_DISABLED_HELP_LINK = helpPagePath(
  'administration/settings/account_and_limit_settings',
  { anchor: 'set-profiles-of-new-users-to-private-by-default' },
);
