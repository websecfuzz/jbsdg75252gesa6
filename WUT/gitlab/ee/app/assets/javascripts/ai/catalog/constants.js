import { __ } from '~/locale';

export const AI_CATALOG_ENUM_AGENT = 'AGENT';
export const AI_CATALOG_NAME_AGENT = __('agent');

export const TYPENAME_AI_CATALOG_ITEM = 'Ai::Catalog::Item';

export const ENUM_TO_NAME_MAP = {
  [AI_CATALOG_ENUM_AGENT]: AI_CATALOG_NAME_AGENT,
};

// Matches backend validations in https://gitlab.com/gitlab-org/gitlab/blob/aa02c3080b316cf0f3b71a992bc5cc5dc8e8bb34/ee/app/models/ai/catalog/item.rb#L10
export const MAX_LENGTH_NAME = 255;
export const MAX_LENGTH_DESCRIPTION = 1024;
export const MAX_LENGTH_PROMPT = 1000000;
