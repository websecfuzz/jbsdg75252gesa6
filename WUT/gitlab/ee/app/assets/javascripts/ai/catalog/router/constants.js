export const AI_CATALOG_INDEX_ROUTE = 'ai-catalog';
export const AI_CATALOG_AGENTS_ROUTE = '/agents';
export const AI_CATALOG_AGENTS_SHOW_ROUTE = '/agents/:id';
export const AI_CATALOG_AGENTS_RUN_ROUTE = '/agents/:id/run';
export const AI_CATALOG_AGENTS_NEW_ROUTE = '/agents/new';
