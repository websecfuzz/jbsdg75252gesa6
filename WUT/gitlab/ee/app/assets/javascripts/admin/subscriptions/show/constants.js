import { __, s__ } from '~/locale';
import { PROMO_URL } from '~/constants';

export const subscriptionMainTitle = s__('SuperSonics|Instance subscription');
export const subscriptionActivationNotificationText = s__(
  `SuperSonics|Your subscription was successfully activated. You can see the details below.`,
);
export const subscriptionActivationFutureDatedNotificationTitle = s__(
  'SuperSonics|Your future dated license was successfully added',
);
export const subscriptionActivationFutureDatedNotificationMessage = s__(
  'SuperSonics|You have successfully added a license that activates on %{date}. Please see the subscription history table below for more details.',
);
export const subscriptionActivationInsertCode = __(
  'Find the activation code for a purchased or renewed subscription in the Customers Portal or purchase confirmation email.',
);
export const howToActivateSubscription = s__(
  'SuperSonics|Learn how to %{linkStart}activate your subscription%{linkEnd}.',
);
export const subscriptionHistoryFailedTitle = s__('SuperSonics|Subscription unavailable');
export const subscriptionHistoryFailedMessage = s__(
  'SuperSonics|Your %{subscriptionEntryName} cannot be displayed at the moment. Please refresh the page to try again.',
);
export const currentSubscriptionsEntryName = s__('SuperSonics|current subscription');
export const pastSubscriptionsEntryName = s__('SuperSonics|past subscriptions');
export const futureSubscriptionsEntryName = s__('SuperSonics|future subscriptions');

export const cancelLabel = __('Cancel');
export const activateLabel = s__('AdminUsers|Activate');
export const activateSubscription = s__('SuperSonics|Activate subscription');
export const addActivationCode = s__('SuperSonics|Add activation code');
export const noActiveSubscription = s__(`SuperSonics|You do not have an active subscription`);
export const subscriptionDetailsHeaderText = s__('SuperSonics|Subscription details');
export const licensedToHeaderText = s__('SuperSonics|Licensed to');
export const copySubscriptionIdButtonText = __('Copy');
export const licenseFileText = __('Legacy license');
export const onlineCloudLicenseText = s__('SuperSonics|Online license');
export const offlineCloudLicenseText = s__('SuperSonics|Offline license');
export const detailsLabels = {
  address: __('Address'),
  company: __('Company'),
  email: __('Email'),
  id: __('ID'),
  lastSync: s__('Subscriptions|Last sync'),
  name: licensedToHeaderText,
  plan: __('Plan'),
  type: __('Type'),
  expiresAt: s__('Subscriptions|End date'),
  startsAt: s__('Subscriptions|Start date'),
};

export const subscriptionTable = {
  activatedAt: s__('Subscriptions|Activation date'),
  expiresOn: s__('Subscriptions|End date'),
  seats: __('Seats'),
  startsAt: s__('Subscriptions|Start date'),
  title: __('Subscription History'),
  type: __('Type'),
};

export const subscriptionActivationForm = {
  activationCode: s__('SuperSonics|Activation code'),
  activationCodeFeedback: s__(
    'SuperSonics|The activation code should be a 24-character alphanumeric string',
  ),
  pasteActivationCode: s__('SuperSonics|Enter activation code'),
  acceptTerms: s__(
    'SuperSonics|I agree that my use of the GitLab Software is subject to the Subscription Agreement located at the %{linkStart}Terms of Service%{linkEnd}, unless otherwise agreed to in writing with GitLab.',
  ),
  acceptTermsFeedback: s__('SuperSonics|Select that you agree before activating the subscription.'),
};

export const subscriptionSyncStatus = {
  SYNC_FAILURE: 'SYNC_FAILURE',
  SYNC_PENDING: 'SYNC_PENDING',
  SYNC_SUCCESS: 'SYNC_SUCCESS',
};

export const subscriptionTypes = {
  ONLINE_CLOUD: 'online_cloud',
  OFFLINE_CLOUD: 'offline_cloud',
  LEGACY_LICENSE: 'legacy_license',
};

export const trialCard = {
  title: s__('SuperSonics|Free trial'),
  description: s__(
    'SuperSonics|You can start a free trial of GitLab Ultimate without any obligation or payment details.',
  ),
  startTrial: s__('SuperSonics|Start free trial'),
};

export const buySubscriptionCard = {
  title: __('Subscription'),
  description: s__(
    'SuperSonics|Ready to get started? A GitLab plan is ideal for scaling organizations and for multi team usage.',
  ),
  buttonLabel: s__('SuperSonics|Buy subscription'),
};

export const SUBSCRIPTION_ACTIVATION_FAILURE_EVENT = 'subscription-activation-failure';
export const SUBSCRIPTION_ACTIVATION_SUCCESS_EVENT = 'subscription-activation-success';
export const SUBSCRIPTION_ACTIVATION_FINALIZED_EVENT = 'subscription-activation-finalized';

// Server-sent error messages (or regexes to match those messages) to detect error type
export const INVALID_ACTIVATION_CODE_SERVER_ERROR = 'invalid activation code';
export const EXPIRED_LICENSE_SERVER_ERROR = 'This license has already expired.';
export const SILENT_MODE_ENABLED_ERROR =
  'only get, head, options, and trace methods are allowed in silent mode';
export const SUBSCRIPTION_NOT_FOUND_SERVER_ERROR = 'without cloud compatible subscription';
export const SUBSCRIPTION_OVERAGES_SERVER_ERROR_REGEX =
  /(This GitLab installation currently has|During the year before this license started, this GitLab installation had) (\d+) active users?, exceeding this license's limit of (\d+) by (\d+) users?\. Please add a license for at least (\d+) users? or contact sales at https:\/\/about\.gitlab\.com\/sales\//;
export const SUBSCRIPTION_INSUFFICIENT_TRUE_UP_SERVER_ERROR_REGEX =
  /You have applied a True-up for (\d+) users? but you need one for (\d+) users?\. Please contact sales at https:\/\/about\.gitlab\.com\/sales\//;

export const CONNECTIVITY_ERROR = 'CONNECTIVITY_ERROR';
export const INVALID_CODE_ERROR = 'INVALID_CODE_ERROR';

export const supportLink = `${PROMO_URL}/support/#contact-support`;
export const exportLicenseUsageBtnText = s__('SuperSonics|Export license usage file');
export const customersPortalBtnText = s__('SuperSonics|Customers Portal');

export const instanceHasFutureLicenseBanner = {
  title: s__('SuperSonics|You have a future dated license'),
  message: s__(
    'SuperSonics|You have added a license that activates on %{date}. Please see the subscription history table below for more details.',
  ),
};

// Subscription Sync Button
export const SYNC_BUTTON_ID = 'syncButton';
export const syncButtonTexts = Object.freeze({
  syncSubscriptionButtonText: s__('SuperSonics|Sync subscription details'),
  syncSubscriptionTooltipText: s__(
    'SuperSonics|You can sync your subscription data to ensure your details are up to date.',
  ),
});

export const VIEW_ADMIN_SUBSCRIPTION_PAGELOAD = 'view_admin_subscription_pageload';
