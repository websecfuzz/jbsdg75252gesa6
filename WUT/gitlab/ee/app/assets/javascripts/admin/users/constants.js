import { s__ } from '~/locale';
import { getTokenConfigs } from '~/admin/users/constants';

export {
  SOLO_OWNED_ORGANIZATIONS_EMPTY,
  I18N_USER_ACTIONS,
  SOLO_OWNED_ORGANIZATIONS_REQUESTED_COUNT,
} from '~/admin/users/constants';

export const TOKEN_CONFIGS = getTokenConfigs([
  { value: 'admins', title: s__('AdminUsers|Administrator') },
  { value: 'auditors', title: s__('AdminUsers|Auditor') },
  { value: 'external', title: s__('AdminUsers|External') },
]);
