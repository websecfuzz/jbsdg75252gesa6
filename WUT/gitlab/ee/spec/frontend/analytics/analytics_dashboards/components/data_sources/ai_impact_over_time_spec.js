import {
  mockAiMetricsResponseData,
  mockAiMetricsZeroResponseData,
  mockAiMetricsNullResponseData,
} from 'ee_jest/analytics/dashboards/ai_impact/mock_data';
import { AI_METRICS } from '~/analytics/shared/constants';
import fetch from 'ee/analytics/analytics_dashboards/data_sources/ai_impact_over_time';
import { defaultClient } from 'ee/analytics/analytics_dashboards/graphql/client';
import {
  LAST_WEEK,
  LAST_30_DAYS,
  LAST_180_DAYS,
} from 'ee/analytics/dora/components/static_data/shared';
import { AI_IMPACT_OVER_TIME_METRICS } from 'ee/analytics/dashboards/ai_impact/constants';

describe('AI Impact Over Time Data Source', () => {
  let res;

  const query = { metric: AI_METRICS.CODE_SUGGESTIONS_USAGE_RATE, dateRange: LAST_30_DAYS };
  const namespace = 'cool namespace';
  const defaultParams = {
    namespace,
    query,
  };

  const mockResolvedQuery = (response = mockAiMetricsResponseData) =>
    jest.spyOn(defaultClient, 'query').mockResolvedValueOnce({ data: { group: response } });

  const expectQueryWithVariables = (variables) =>
    expect(defaultClient.query).toHaveBeenCalledWith(
      expect.objectContaining({
        variables: expect.objectContaining(variables),
      }),
    );

  describe('fetch', () => {
    describe('default', () => {
      it('correctly applies query parameters', async () => {
        await mockResolvedQuery();
        res = await fetch({ namespace, query });

        expectQueryWithVariables({
          startDate: new Date('2020-06-07'),
          endDate: new Date('2020-07-07'),
          fullPath: namespace,
        });
      });

      describe.each`
        type             | response                         | result
        ${'valid'}       | ${mockAiMetricsResponseData}     | ${'62.5'}
        ${'zero values'} | ${mockAiMetricsZeroResponseData} | ${'-'}
        ${'null values'} | ${mockAiMetricsNullResponseData} | ${'-'}
      `('with $type data', ({ response, result }) => {
        beforeEach(async () => {
          mockResolvedQuery(response);
          res = await fetch({ namespace, query });
        });

        it(`returns ${result}`, () => {
          expect(res).toBe(result);
        });
      });
    });

    describe('setVisualizationOverrides callback', () => {
      describe.each(Object.keys(AI_IMPACT_OVER_TIME_METRICS))('for %s metric', (metric) => {
        let mockSetVisualizationOverrides;

        describe.each`
          description                    | response
          ${'with valid response'}       | ${mockAiMetricsResponseData}
          ${'with null values response'} | ${mockAiMetricsNullResponseData}
          ${'with zeroes response'}      | ${mockAiMetricsZeroResponseData}
        `('$description', ({ response }) => {
          beforeEach(async () => {
            mockSetVisualizationOverrides = jest.fn();

            mockResolvedQuery(response);
            res = await fetch({
              namespace,
              query: { ...defaultParams.query, metric },
              setVisualizationOverrides: mockSetVisualizationOverrides,
            });
          });

          it('will call the setVisualizationOverrides callback with the correct settings', () => {
            expect(mockSetVisualizationOverrides.mock.calls).toMatchSnapshot();
          });
        });
      });
    });

    describe('queryOverrides', () => {
      const mockQuery = (
        dateRange,
        {
          namespace: namespaceParam = 'cool namespace',
          metric = AI_METRICS.CODE_SUGGESTIONS_USAGE_RATE,
        } = {},
      ) => {
        mockResolvedQuery();

        return fetch({
          ...defaultParams,
          query: { ...defaultParams.query, metric },
          queryOverrides: { dateRange, namespace: namespaceParam },
        });
      };

      it('can override the date range', async () => {
        res = await mockQuery(LAST_WEEK);

        expectQueryWithVariables({
          startDate: new Date('2020-06-30'),
          endDate: new Date('2020-07-07'),
          fullPath: namespace,
        });
      });

      it.each`
        metric                                         | result
        ${AI_METRICS.CODE_SUGGESTIONS_USAGE_RATE}      | ${'62.5'}
        ${AI_METRICS.CODE_SUGGESTIONS_ACCEPTANCE_RATE} | ${'40.0'}
        ${AI_METRICS.DUO_CHAT_USAGE_RATE}              | ${'50.0'}
        ${AI_METRICS.DUO_USAGE_RATE}                   | ${'30.0'}
      `('can override the metric with `$metric`', async ({ metric, result }) => {
        res = await mockQuery(LAST_WEEK, { metric });

        expectQueryWithVariables({
          startDate: new Date('2020-06-30'),
          endDate: new Date('2020-07-07'),
          fullPath: namespace,
        });

        expect(res).toBe(result);
      });

      it('can override the namespace', async () => {
        res = await mockQuery(LAST_WEEK, { namespace: 'cool-namespace/sub-namespace' });

        expectQueryWithVariables({
          startDate: new Date('2020-06-30'),
          endDate: new Date('2020-07-07'),
          fullPath: 'cool-namespace/sub-namespace',
        });
      });

      it('will default to LAST_180_DAYS when given an invalid dateRange', async () => {
        res = await mockQuery('LAST_45_DAYS');

        expectQueryWithVariables({
          startDate: new Date('2020-01-09'),
          endDate: new Date('2020-07-07'),
          fullPath: namespace,
        });

        const defaultRes = await mockQuery(LAST_180_DAYS);
        expect(defaultRes).toEqual(res);
      });
    });
  });
});
