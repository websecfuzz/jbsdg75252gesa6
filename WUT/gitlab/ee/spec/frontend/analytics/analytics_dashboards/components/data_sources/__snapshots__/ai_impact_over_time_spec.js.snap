// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_acceptance_rate metric with null values response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total number of code suggestions generated / code suggestions accepted. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_acceptance_rate metric with valid response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total number of code suggestions generated / code suggestions accepted. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_acceptance_rate metric with zeroes response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total number of code suggestions generated / code suggestions accepted. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_usage_rate metric with null values response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Monthly user engagement with AI Code Suggestions. Percentage ratio calculated as monthly unique Code Suggestions users / total monthly unique code contributors in the last 30 days. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_usage_rate metric with valid response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Monthly user engagement with AI Code Suggestions. Percentage ratio calculated as monthly unique Code Suggestions users / total monthly unique code contributors in the last 30 days. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for code_suggestions_usage_rate metric with zeroes response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Monthly user engagement with AI Code Suggestions. Percentage ratio calculated as monthly unique Code Suggestions users / total monthly unique code contributors in the last 30 days. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/project/repository/code_suggestions/_index#use-code-suggestions",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_chat_usage_rate metric with null values response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that interacted with Duo Chat. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/gitlab_duo_chat/_index",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_chat_usage_rate metric with valid response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that interacted with Duo Chat. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/gitlab_duo_chat/_index",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_chat_usage_rate metric with zeroes response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that interacted with Duo Chat. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/user/gitlab_duo_chat/_index",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_usage_rate metric with null values response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that used at least one Duo feature. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/subscriptions/subscription-add-ons#assign-gitlab-duo-seats",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_usage_rate metric with valid response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that used at least one Duo feature. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/subscriptions/subscription-add-ons#assign-gitlab-duo-seats",
        },
      },
    },
  ],
]
`;

exports[`AI Impact Over Time Data Source fetch setVisualizationOverrides callback for duo_usage_rate metric with zeroes response will call the setVisualizationOverrides callback with the correct settings 1`] = `
[
  [
    {
      "visualizationOptionOverrides": {
        "title": "Last 30 days",
        "tooltip": {
          "description": "Percentage ratio of total Duo seats assigned / Duo seats assigned that used at least one Duo feature. %{linkStart}Learn more%{linkEnd}.",
          "descriptionLink": "/help/subscriptions/subscription-add-ons#assign-gitlab-duo-seats",
        },
      },
    },
  ],
]
`;
